### PR TITLE
FWE: Handling user input (v1)

### PR DESCRIPTION
**Pull request by khanhnwin**
_Description of what this PR is changing or adding, and why:_ The initial "Handling user input" page for FWE v1

### CODE REVIEW

File: src/get-started/fwe/user-input.md [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+[Form]: {{site.api}}flutter/widgets/Form-class.html
+[Build a form with validation]: /cookbook/forms/validation
+[Form App]: https://flutter.github.io/samples/web/form_app/
+[Form App Code]: https://github.com/flutter/samples/tree/main/form_app
+[SelectableText]: {{site.api}}flutter/material/SelectableText-class.html
+[SelectableText (Widget of the Week)]: https://www.youtube.com/watch?v=ZSU3ZXOs6hc
+
+[SegmentedButton]: {{site.api}}flutter/material/SegmentedButton-class.html
+[DropdownMenu]: {{site.api}}flutter/material/DropdownMenu-class.html
+[DropdownMenu (Widget of the Week)]: https://youtu.be/giV9AbM2gd8?si=E23hjg72cjMTe_mz
```
Comment (sfshaza2) [inline src/get-started/fwe/user-input.md:183]:
```suggestion
[`TextField`]: {{site.api}}flutter/material/TextField-class.html
[Create and style a text field]: /cookbook/forms/text-input
[Retrieve the value of a text field]: /cookbook/forms/retrieve-input
[Handle changes to a text field]: /cookbook/forms/text-field-changes
[Focus and text fields]: /cookbook/forms/focus
[`RichText`]: {{site.api}}flutter/widgets/RichText-class.html
[Rich Text (Widget of the Week)]: https://www.youtube.com/watch?v=rykDVh-QFfw
[Rich Text Editor]: https://flutter.github.io/samples/rich_text_editor.html
[Rich Text Editor code]: https://github.com/flutter/samples/tree/main/simplistic_editor
[`Form`]: {{site.api}}flutter/widgets/Form-class.html
[Build a form with validation]: /cookbook/forms/validation
[Form App]: https://flutter.github.io/samples/web/form_app/
[Form App Code]: https://github.com/flutter/samples/tree/main/form_app
[`SelectableText`]: {{site.api}}flutter/material/SelectableText-class.html
[SelectableText (Widget of the Week)]: https://www.youtube.com/watch?v=ZSU3ZXOs6hc

[`SegmentedButton`]: {{site.api}}flutter/material/SegmentedButton-class.html
[`DropdownMenu`]: {{site.api}}flutter/material/DropdownMenu-class.html
[DropdownMenu (Widget of the Week)]: https://youtu.be/giV9AbM2gd8?si=E23hjg72cjMTe_mz
```

File: src/get-started/fwe/user-input.md [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+[Dismissible]: {{site.api}}flutter/widgets/Dismissible-class.html
+[Dismissible (Widget of the Week)]: https://youtu.be/iEMgjrfuc58?si=f0S7IdaA9PIWIYvl
+[Implement swipe to dismiss]: /cookbook/gestures/dismissible
+[pkg:flutter_slidable]: https://pub.dev/packages/flutter_slidable
+[flutter_slidable (Package of the Week)]: https://www.youtube.com/watch?v=QFcFEpFmNJ8
+
+[GestureDetector (Widget of the Week)]: https://www.youtube.com/watch?v=WhVXkCFPmK4
+[Taps, drags, and other gestures]: /ui/interactivity/gestures#gestures
+[Handle taps]: /cookbook/gestures/handling-taps
+[Semantics (Flutter Widget of the Week)]: https://youtu.be/NvtMt_DtFrQ?si=o79BqAg9NAl8EE8_
```
Comment (sfshaza2) [inline src/get-started/fwe/user-input.md:211]:
```suggestion
[`Dismissible`]: {{site.api}}flutter/widgets/Dismissible-class.html
[Dismissible (Widget of the Week)]: https://youtu.be/iEMgjrfuc58?si=f0S7IdaA9PIWIYvl
[Implement swipe to dismiss]: /cookbook/gestures/dismissible
[pkg:`flutter_slidable`]: https://pub.dev/packages/flutter_slidable
[flutter_slidable (Package of the Week)]: https://www.youtube.com/watch?v=QFcFEpFmNJ8

[GestureDetector (Widget of the Week)]: https://www.youtube.com/watch?v=WhVXkCFPmK4
[Taps, drags, and other gestures]: /ui/interactivity/gestures#gestures
[Handle taps]: /cookbook/gestures/handling-taps
[Semantics (Flutter Widget of the Week)]: https://youtu.be/NvtMt_DtFrQ?si=o79BqAg9NAl8EE8_
```

**Review by sfshaza2 — COMMENTED** [review]
Thanks for doing this on such short notice, @khanhnwin (well, kinda sorta), and when you're swamped with I/O. I made quite a few edits, but just accepting at least some of them should make your life easier...

**Review by sfshaza2 — CHANGES_REQUESTED** [review]
Changing status to request changes, since Brett already approved.

**Comment by sfshaza2** [comment]
Also, Khanh, that first section you use bulleted lists for the widgets. But in subsequent sections, you don't. I haven't looked at the generated page yet, but please be consistent. thx!

File: src/get-started/fwe/user-input.md [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+
+Now that you know how to manage state in your Flutter app, how can you let users interact with your app and change its state? 
+
+## Introduction to user input
+
+As a multi-platform UI framework, there are many different ways for users to interact with a Flutter app. The resources in this section will introduce you to some of the common widgets used for enabling user interaction. 
+
+Some user input types, like [scrolling], have already been covered in previous sections. 
+
+For an introduction to how you can change app state in response to user interactions, check out the following tutorial which teaches you how to build a "favorite" button:
```
[the remark this answers is not in the document]
Comment (khanhnwin) [reply src/get-started/fwe/user-input.md:20]: I love whitespace! I will agree to this revision anyway XD

File: src/get-started/fwe/user-input.md [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+
+<i class="material-symbols" aria-hidden="true">flutter_dash</i> Tutorial: [Add interactivity to your Flutter app][]
+
+## Material 3 demo 
+
+Now that you've gotten a feel for how user interaction works, check out the [Material 3 Demo][] which samples some of the various user input widgets available in the Material 3 component library.
```
[the remark this answers is not in the document]
Comment (khanhnwin) [reply src/get-started/fwe/user-input.md:27]: TIL!

File: src/get-started/fwe/user-input.md [hunk 5]
```diff
@@ -0,0 +1,215 @@
...
+
+For an introduction to how you can change app state in response to user interactions, check out the following tutorial which teaches you how to build a "favorite" button:
+
+<i class="material-symbols" aria-hidden="true">flutter_dash</i> Tutorial: [Add interactivity to your Flutter app][]
+
+## Material 3 demo 
+
+Now that you've gotten a feel for how user interaction works, check out the [Material 3 Demo][] which samples some of the various user input widgets available in the Material 3 component library. 
+
+Note: This page focuses on Material 3 widgets because it is the library that many developers start with when learning Flutter. However, Flutter also supports other component libraries such as [Cupertino][] for iOS-styled widgets.
```
[the remark this answers is not in the document]
Comment (khanhnwin) [reply src/get-started/fwe/user-input.md:31]: There are other packages available too. Ex: [macos_ui](https://pub.dev/packages/macos_ui)

File: src/get-started/fwe/user-input.md [hunk 6]
```diff
@@ -0,0 +1,215 @@
...
+* [ElevatedButton][]: A button with some depth. Use elevated buttons to add dimension to otherwise mostly flat layouts.
+* [FilledButton][]: A filled button that should be used for important, final actions that complete a flow, like Save, Join now, or Confirm.
+* [TextButton][]: Clickable text, without a border. 
+* [OutlinedButton][]: A button with text and a visible border.
+
+[IconButton][]: A button with an icon. 
+
+[FloatingActionButton][]: A circular icon button that hovers over content to promote a primary action.
+
+* Video: [FloatingActionButton (Widget of the Week)][]
```
[the remark this answers is not in the document]
Comment (khanhnwin) [reply src/get-started/fwe/user-input.md:58]: They're all technically buttons, but not the same type of "core" buttons.

[the remark this answers is not in the document]
Comment (khanhnwin) [reply src/get-started/fwe/user-input.md:64]: I added them to the list!

**Comment by domesticmouse** [comment]
Both the GitHub markdown preview and the link checker don't like this change:

```console
Checking for broken Markdown link references...
Error: Invalid link references found!

get-started/fwe/user-input/index.html:
[TextField][]
```

**Comment by khanhnwin** [comment]
ptal @sfshaza2!

**Review by sfshaza2 — APPROVED** [review]
Thanks, @khanhnwin! There are still some formatting issues, but I'm going to land this and fix them in my index PR.

**Comment by khanhnwin** [comment]
Thanks @sfshaza2!

**Pull request merged**